src/dim_helpers/ConvDims.jl

```diff
@@ -6,34 +6,54 @@ export ConvDims
 Type system-level information about convolution dimensions. Critical for things like
 `im2col!()` to generate efficient code, and helpful to reduce the number of kwargs
 getting passed around.
-
-We don't want to specialize on things like image size/channel count, so we generally
-store those as fields, just for convenience, and to allow for non-breaking changes when
-we decide we _do_ want to specialize on those values.  We always want to specialize on
-things like stride, padding, dilation, and kernel flipping though.
 """
-abstract type ConvDims{N, S, P, D, F} end
+abstract type ConvDims{N} end
+
+@inline spatial_dims(::ConvDims{N}) where N = N
+@inline input_size(c::ConvDims) = c.input_size
+@inline kernel_size(c::ConvDims) = c.kernel_size
+
+@inline stride(c::ConvDims) = c.stride
+@inline padding(c::ConvDims) = c.padding
+@inline dilation(c::ConvDims) = c.dilation
+@inline flipkernel(c::ConvDims) = c.flipkernel
```

Comment:
I've [tried](https://github.com/pxl-th/NNlib.jl/blob/fix-perf/src/impl/conv_im2col.jl#L8) adding `flipkernel` value to the type, not boolean like it was before, but `Val{true}` to keep type-stability.
However, the performance was pretty much the same. Maybe because previously the kernel size was also part of the type and now it has to [retrieve](https://github.com/FluxML/NNlib.jl/blob/master/src/impl/conv_im2col.jl#L6) this property every time from the struct.

Reply:
Confirmed on my side as well, there's basically no difference with master so I think this was a dead end. Feel free to revert.